### PR TITLE
set initial system time from RTC fixed, time-valid added HYP-21732

### PR DESCRIPTION
**Pull request by dierberg**

### CODE REVIEW

File: recipes-hipos/hipos-time/files/set-time.sh [hunk 1]
```diff
@@ -1,19 +1,61 @@
...
-MACHINE_TYPE="$(/usr/sbin/hip-machinfo -a)"
+log() {
+	logger -t set-time.sh $1
+}
+
 SUB_TYPE="$(/usr/sbin/hip-machinfo -s)"
-VARIANT=$(. /etc/hydraip-devid; echo "${device:-recorder}")
+
+# reset error signaling
+rm /etc/drerrors/RTC.json 2> /dev/null
```
Comment (sledz) [inline recipes-hipos/hipos-time/files/set-time.sh]: I'd prefer `rm -f /etc/drerrors/RTC.json`

File: recipes-hipos/hipos-time/files/set-time.sh [hunk 2]
```diff
@@ -1,19 +1,61 @@
...
+# got last shutdown time
+last=$(cat /lib/systemd/systemd-timesyncd 2> /dev/null)
+
+# $date and $last may be empty or invalid here
+if [ -n "$date" ]; then
+	# empty battery
+	if [[ "$date" < "2019" ]]; then
+		log "Warning: RTC time outdated: $date"
+		# signal error with LED
+		echo "{\"error\":\"RTC time outdated: $date\"}" > /etc/drerrors/RTC.json
```
Comment (sledz) [inline recipes-hipos/hipos-time/files/set-time.sh:37]: It isn't save to assume that directory `/etc/drerrors` still exists.

File: recipes-hipos/hipos-time/files/set-time.sh [hunk 3]
```diff
@@ -1,19 +1,61 @@
...
 
+# got last shutdown time
+last=$(cat /lib/systemd/systemd-timesyncd 2> /dev/null)
+
+# $date and $last may be empty or invalid here
+if [ -n "$date" ]; then
+	# empty battery
+	if [[ "$date" < "2019" ]]; then
+		log "Warning: RTC time outdated: $date"
+		# signal error with LED
```
Comment (sledz) [inline recipes-hipos/hipos-time/files/set-time.sh:35]: Does HIPOS really support this kind of LED controlling?

**Review by sledz — COMMENTED** [review]
Please provide a more helpful commit comment!

File: recipes-hipos/hipos-time/files/set-time.sh (hunk 1 above)
Comment (dierberg) [reply recipes-hipos/hipos-time/files/set-time.sh]: Ack

File: recipes-hipos/hipos-time/files/set-time.sh (hunk 2 above)
Comment (dierberg) [reply recipes-hipos/hipos-time/files/set-time.sh:37]: What do you suggest? " 2> /dev/null "?

Comment (sledz) [reply recipes-hipos/hipos-time/files/set-time.sh:37]: `mkdir -p /etc/drerrors` before writing to the file.

File: recipes-hipos/hipos-time/files/set-time.sh (hunk 3 above)
Comment (dierberg) [reply recipes-hipos/hipos-time/files/set-time.sh:35]: No, they can ignore this

File: recipes-hipos/hipos-time/files/set-time.sh (hunk 2 above)
Comment (dierberg) [reply recipes-hipos/hipos-time/files/set-time.sh:37]: Ack

File: recipes-hipos/hipos-time/files/set-time.sh [hunk 4]
```diff
@@ -1,19 +1,62 @@
...
-                done 2>&1 | logger -t $0
-                set -x
-        }
-else
-        date -s `cat /lib/systemd/systemd-timesyncd` 2> /dev/null        
+	date=$(drbcc --dev=/dev/ttydrbcc --cmd=getrtc | cut -d\  -f2,3)
+	if [ -z "$date" ]; then
+		# in case of problems we log the commands
+		set -x
+		for try in 1 2 3 4 5; do
```
Comment (reikowski) [inline recipes-hipos/hipos-time/files/set-time.sh:17]: My IDE's ShellCheck extension says "try appears unused. Verify use (or export if used externally).shellcheck(SC2034)" and more.

You can "shellcheck" your code using https://www.shellcheck.net/

Click on the SCxxxx links there for further explanation.

**Review by marioecht — APPROVED** [review]

File: recipes-hipos/hipos-time/files/set-time.sh (hunk 4 above)
Comment (dierberg) [reply recipes-hipos/hipos-time/files/set-time.sh:17]: Yes it seems to be unused, it is the retry counter.

**Pull request merged**